File: data/imported/knowledge/slides/java/merged/snippets/snippet_052.java
@Test
    void testLineDashStyle() throws IOException {
        try (var pres = new Presentation()) {
            var slide = clear(pres);
            var shape = slide.getShapes().addAutoShape(ShapeType.RECTANGLE, 50, 50, 200, 100);
            var lf = shape.getLineFormat();
            lf.setWidth(3);
            lf.setDashStyle(LineDashStyle.DASH);
            lf.getFillFormat().setFillType(FillType.SOLID);
            lf.getFillFormat().getSolidFillColor().setColor(Color.BLACK);

            try (var pres2 = saveAndReopen(pres)) {
                var lf2 = pres2.getSlides().get(0).getShapes().get(0).getLineFormat();
                assertThat(lf2.getDashStyle()).isEqualTo(LineDashStyle.DASH);
            }
        }
    }

File: data/imported/knowledge/slides/cpp/merged/snippets/snippet_052.cpp
TEST_F(ShapesIntegrationTest, RemoveShape) {
    Presentation pres;
    auto& slide = blank_slide(pres);
    auto& s = slide.shapes().add_auto_shape(ShapeType::RECTANGLE, 50, 50, 200, 100);
    slide.shapes().add_auto_shape(ShapeType::ELLIPSE, 300, 50, 150, 150);
    EXPECT_EQ(slide.shapes().size(), 2);
    slide.shapes().remove(s);
    EXPECT_EQ(slide.shapes().size(), 1);
}